pkg/codegen/schema/bind.go

```diff
@@ -1076,6 +1103,11 @@ func (t *types) bindProperties(path string, properties map[string]PropertySpec,
 			return nil, nil, diags, fmt.Errorf("error binding type for property %q: %w", name, err)
 		}
 
+		// After binding the type of a property we need to check if it's overly nested, we can only handle two layers of
+		// nesting so array[array[string]] is ok but array[array[array[string]]] is not.
+		diags = append(diags, isOverlyNested(propertyPath, typ)...)
```

Comment:
It took me a moment to spot the `...` and I was wondering if we were adding a `nil` value to the slice here. Maybe `diags.Extend` would make it clearer?